guides/plugins/apps/platform-sh-template.md

```diff
@@ -0,0 +1,208 @@
+
+## The argument resolver
+
+There are two argument resolver. One for the [Client](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Client.php) and one for the [Event](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Event.php).  
+The purpose of those is to inject the [Client](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Client.php) and the [Event](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Event.php) whenever you need them.  
+
+For example you define a route for incoming webhooks and want to fetch some extra data. Then you can use them as a parameter of the method which will be called when a request is send to the route.  
+
+But how do you know that the request is from the shop and not from someone who is sending post requests to your app?
+The argument resolver take care of it. Whenever you use one of them as a parameter the request will be authenticated. If the request isn't authenticated the [Client](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Client.php) or the [Event](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Event.php) will be null. 
+
+## The shop repository
+
+The [ShopRepository](https://github.com/shopwareLabs/AppTemplate/blob/master/src/Repository/ShopRepository.php) can be used to get the secret of the shop and the [Credentials](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Credentials.php).  
+
+For example if you want to build your own authentication you can use the [ShopRepository](https://github.com/shopwareLabs/AppTemplate/blob/master/src/Repository/ShopRepository.php) to get the secret to the corresponding shop.  
+But if you want to build your own [Client](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Client.php) you can simply get the [Credentials](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/Credentials.php) for a specific `shopId`.  
+ 
+## App lifecycle events
+
+There are five app lifecycle events which can be triggered during the lifecycle of an app. The events are `app.installed`, `app.updated`, `app.deleted`, `app.activated` and `app.deactivated`.
+To use this events you have to create the webhooks in your manifest. If you want to implement your own code you only need to implement the [AppLifecycleHandler](https://github.com/shopwareLabs/AppTemplate/blob/master/src/SwagAppsystem/AppLifecycleHandler.php) interface and write your own code.  
+
+The `app.installed` event gets triggered each time the app gets installed. This will also trigger the `app.activated` event. At each of this both events the shop is already installed and registered at your app. The webhook could look like this:
+```xml
+<webhook name="appLifecycleInstalled" url="https://your-shop-url/applifecycle/installed" event="app.installed"/>
+```
```

Comment:
Is this the manifest.xml ? 
If so please keep the "title", otherwise remove it

``````suggestion
{% code title="manifest.xml" %}
```xml
<webhook name="appLifecycleInstalled" url="https://your-shop-url/applifecycle/installed" event="app.installed"/>
```
{% endcode %}
``````

Reply:
Yes this is part of the `manifest.xml`. I will keep the "title" :smile: